### PR TITLE
GH-1623 fix typo for updating node in Neo4jVectorStore

### LINKED ISSUE #1623: if the embedding property of a node is not using the default name, documents cannot be added to the vector store
Please do a quick search on GitHub issues first, there might be already a duplicate issue for the one you are about to create.
If the bug is trivial, just go ahead and create the issue. Otherwise, please take a few moments and fill in the following sections:

**Bug description**
See the tile
**Environment**
Please provide as many details as possible: Spring AI version, Java version, which vector store you use if any, etc
Spring AI 1.0.0-M3
Java 21
Vector store: Neo4J 5.24.1
**Steps to reproduce**
In your application.properties set the property `spring.ai.vectorstore.neo4j.embedding-property` with the name of the embedding property you want to use and add a document to the vector store
**Expected behavior**
The document is added to the vector store
**Minimal Complete Reproducible example**
See the attached zip file
[spring ai neo4j vector store bug.zip](https://github.com/user-attachments/files/17561228/spring.ai.neo4j.vector.store.bug.zip)

**Solution**
 In https://github.com/spring-projects/spring-ai/blob/ca1949c8250187d94d735665b2044a844234aa51/vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java#L120, change `row.embedding` to `row[$embeddingProperty]`

I would be more than happy to contribute this bugfix, let me know if you want me to :)

### PR DESCRIPTION
**Pull request by jitokim**
- Resolve #1623 
- Define DEFAULT_TRANSACTION_SIZE for the hardcoded batch size in the delete query

### CODE REVIEW

**Comment by tzolov** [comment]
Hi @michael-simons , can you please review this PR?

File: vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java [hunk 1]
```diff
@@ -132,8 +135,8 @@ public Optional<Boolean> doDelete(List<String> idList) {
 				.run("""
 						MATCH (n:%s) WHERE n.%s IN $ids
 						CALL { WITH n DETACH DELETE n } IN TRANSACTIONS OF $transactionSize ROWS
-						""".formatted(this.config.label, this.config.idProperty),
-						Map.of("ids", idList, "transactionSize", 10_000))
+						 """.formatted(this.config.label, this.config.idProperty),
```
Comment (michael-simons) [inline vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java:138]:
```suggestion
						""".formatted(this.config.label, this.config.idProperty),
```

**Review by michael-simons — APPROVED** [review]
Nice change, also with the tx size. Added one small change request, though (OCD kicking in…)

File: vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java (hunk 1 above)
Comment (jitokim) [reply vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java]: @michael-simons  Thanks for review 👍

Comment (michael-simons) [reply vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/vectorstore/Neo4jVectorStore.java]: ofc.

**Comment by jitokim** [comment]
@michael-simons  thanks for review 😀

**Comment by markpollack** [comment]
thanks for the contribution and the review!

merged in 1cdec7b6c3f7c15d0c1b741b9fce09be93105d86

**Pull request closed without merging**